src/controllers/expense.controller.js

```diff
@@ -0,0 +1,92 @@
+const expenseService = require('../services/expense.service.js');
+const userService = require('../services/user.service.js');
+
+const get = async (req, res) => {
+  const { categories, userId, from, to } = req.query;
+
+  const result = await expenseService.getAll(categories, userId, from, to);
+
+  return res.send(result);
+};
+
+const getById = async (req, res) => {
+  const { id } = req.params;
+
+  const expense = await expenseService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const create = async (req, res) => {
+  const { userId, title, amount, category, note, spentAt } = req.body;
+  const existedUser = await userService.getById(userId);
+
+  if (!userId || !title || !amount || !spentAt || !existedUser) {
+    return res.sendStatus(400);
+  }
+
+  const newExpense = await expenseService.createExpense({
+    userId,
+    title,
+    amount,
+    spentAt: new Date(spentAt),
+    category,
+    note,
+  });
+
+  res.statusCode = 201;
+
+  return res.send(newExpense);
+};
+
+const remove = async (req, res) => {
+  const { id } = req.params;
+  const expense = await expenseService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  await expenseService.remove(id);
+
+  res.sendStatus(204);
+};
+
+const update = async (req, res) => {
+  const data = req.body;
+  const { id } = req.params;
+
+  if (!id) {
+    res.sendStatus(404);
+
+    return;
```

Comment:
The check for `!id` seems redundant because `id` is extracted from `req.params`. If `id` is not present, it would typically result in a 404 error earlier in the request lifecycle. Consider removing this check or handling it in middleware.